dts/arm/nxp/nxp_mcxn23x_common.dtsi

```diff
@@ -0,0 +1,692 @@
+			status = "disabled";
+		};
+		flexcomm5_lpi2c5: lpi2c@b5800 {
+			compatible = "nxp,imx-lpi2c";
+			reg = <0xb5800 0x1000>;
+			clocks = <&syscon MCUX_FLEXCOMM5_CLK>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+	};
+
+	flexcomm6: flexcomm@b6000 {
+		compatible = "nxp,lp-flexcomm";
+		reg = <0xb6000 0x1000>;
+		interrupts = <41 0>;
+		status = "disabled";
+
+		ranges = <>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		flexcomm6_lpuart6: lpuart@b6000 {
+			compatible = "nxp,kinetis-lpuart";
+			reg = <0xb6000 0x1000>;
+			clocks = <&syscon MCUX_FLEXCOMM6_CLK>;
+			status = "disabled";
+		};
+		flexcomm6_lpspi6: lpspi@b6000 {
+			compatible = "nxp,imx-lpspi";
+			reg = <0xb6000 0x1000>;
+			clocks = <&syscon MCUX_FLEXCOMM6_CLK>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+		flexcomm6_lpi2c6: lpi2c@b6800 {
+			compatible = "nxp,imx-lpi2c";
+			reg = <0xb6800 0x1000>;
+			clocks = <&syscon MCUX_FLEXCOMM6_CLK>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+	};
+
+	flexcomm7: flexcomm@b7000 {
+		compatible = "nxp,lp-flexcomm";
+		reg = <0xb7000 0x1000>;
+		interrupts = <42 0>;
+		status = "disabled";
+
+		ranges = <>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		flexcomm7_lpuart7: lpuart@b7000 {
+			compatible = "nxp,kinetis-lpuart";
+			reg = <0xb7000 0x1000>;
+			clocks = <&syscon MCUX_FLEXCOMM7_CLK>;
+			status = "disabled";
+		};
+		flexcomm7_lpspi7: lpspi@b7000 {
+			compatible = "nxp,imx-lpspi";
+			reg = <0xb7000 0x1000>;
+			clocks = <&syscon MCUX_FLEXCOMM7_CLK>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+		flexcomm7_lpi2c7: lpi2c@b7800 {
+			compatible = "nxp,imx-lpi2c";
+			reg = <0xb7800 0x1000>;
+			clocks = <&syscon MCUX_FLEXCOMM7_CLK>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+	};
+
+	edma0: dma-controller@80000 {
+		#dma-cells = <2>;
+		compatible = "nxp,mcux-edma-v4";
+		dma-channels = <16>;
+		dma-requests = <120>;
+
+		reg = <0x80000 0x1000>;
+		interrupts = <1 0>, <2 0>, <3 0>, <4 0>,
+			<5 0>, <6 0>, <7 0>, <8 0>,
+			<9 0>, <10 0>, <11 0>, <12 0>,
+			<13 0>, <14 0>, <15 0>, <16 0>;
+		no-error-irq;
+		status = "disabled";
+	};
+
+	edma1: dma-controller@a0000 {
+		#dma-cells = <2>;
+		compatible = "nxp,mcux-edma-v4";
+		dma-channels = <16>;
+		dma-requests = <120>;
+
+		reg = <0xa0000 0x1000>;
+		interrupts = <77 0>, <78 0>, <79 0>, <80 0>,
+			<81 0>, <82 0>, <83 0>, <84 0>,
+			<85 0>, <86 0>, <87 0>, <88 0>,
+			<89 0>, <90 0>, <91 0>, <92 0>;
+		no-error-irq;
+		status = "disabled";
+	};
+
+	fmu: flash-controller@43000 {
+		compatible = "nxp,iap-msf1";
+		reg = <0x43000 0x1000>;
+		interrupts = <138 0>;
+		status = "disabled";
+
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		flash: flash@0 {
+			compatible = "soc-nv-flash";
+			reg = <0 DT_SIZE_M(1)>;
+			erase-block-size = <8192>;
+			write-block-size = <16>;
```

Comment:
I know for the MCXN94x that the FMU itself supports 16 byte writes, but the ROM API the flash driver uses only supports 128 byte writes. Is that the case for this SOC as well?

Reply:
About FMU, MCXN23x should be same as MCXN94x.

Reply:
> About FMU, MCXN23x should be same as MCXN94x.

According to the MCXN23xRM documentation, the Flash ROM API supports only 128 Bytes. So the write-block-size must be set to 128, as for MCXN94x.